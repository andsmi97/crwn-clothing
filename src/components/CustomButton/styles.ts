import styled, { css } from "styled-components";

const invertedButtonStyles = css`
  background-color: white;
  color: black;
  border: 1px solid black;

  &:hover {
    background-color: black;
    color: white;
    border: none;
  }
`;

const googleSignInStyles = css`
  background-color: #4285f4;
  color: #fff;

  &:hover {
    background-color: #357ae8;
    border: none;
  }
`;
const buttonStyles = css`
  background-color: black;
  color: white;
  border: none;
  &:hover {
    background-color: white;
    color: black;
    border: 1px solid black;
  }
`;

const getButtonStyles = (props: any) => {
  if (props.isGoogleSignIn) {
    return googleSignInStyles;
  }
  return props.inverted ? invertedButtonStyles : buttonStyles;
};

export const CustomButtonContainer = styled.button`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 15px;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: "Open Sans Condensed";
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;

  ${getButtonStyles}
`;
