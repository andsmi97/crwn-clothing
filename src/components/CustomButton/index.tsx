import React from "react";

import "./styles.scss";
import { ICustomButton } from "../../types";

export default ({
  children,
  isGoogleSignIn,
  ...restProps
}: ICustomButton): JSX.Element => {
  return (
    <button
      className={`${isGoogleSignIn ? "google-sign-in" : ""} custom-button`}
      {...restProps}
    >
      {children}
    </button>
  );
};
