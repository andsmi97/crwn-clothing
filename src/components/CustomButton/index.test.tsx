import React from "react";
import { shallow } from "enzyme";
import CustomButton from "./index";

describe("<CustomButton />", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<CustomButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
