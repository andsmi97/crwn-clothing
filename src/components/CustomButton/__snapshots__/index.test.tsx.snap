// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> matches snapshot 1`] = `ShallowWrapper {}`;
